### DISCUSSION TITLE
error PermissionError: [Errno 13] Permission denied: 'models/7B//ggml-model-f16.bin' with the command "python3 convert-pth-to-ggml.py models/7B/1"

### DISCUSSION THREAD

**Discussion by MarcGach** [ggml-org/llama.cpp#556] [Q&A]
Hi everyone,
I am near the end of the procedure, at the time of making this command,
`python3 convert-pth-to-ggml.py models/7B/1`

I get this error:
```
Traceback (most recent call last):
  File "/opt/openai/llama.cpp/convert-pth-to-ggml.py", line 181, in <module>
    hand()
  File "/opt/openai/llama.cpp/convert-pth-to-ggml.py", line 171, in main
    with open(fname_out, "wb") as fout:
PermissionError: [Errno 13] Permission denied: 'models/7B//ggml-model-f16.bin'
```
 
The content of my model folder, contains the 7B, Tokenizer.model that I got from the internet, there is only the tokenizer_checklist.chk where I copied and pasted the content on a site.

I'm struggling a bit and I thank you for your answers if you have already encountered this problem.

**Comment by anzz1** [comment]
The correct command is:
`python3 convert-pth-to-ggml.py models/7B 1`
You have a `/` between `7B` and `1`, there should be a space.

It sounds like you have the paths correct, just the command is wrong.
Your path should look like this:
`models/tokenizer.model`
`models/7B/consolidated.00.pth`
`models/7B/params.json`

Those are the files you need , the checklist file isn't required its extra and contains checksums.

Afterwards, you can quantize it to either q4_0 or q4_1 to make it smaller.

`./quantize ./models/7B/ggml-model-f16.bin ./models/7B/ggml-model-q4_0.bin 2` for q4_0
`./quantize ./models/7B/ggml-model-f16.bin ./models/7B/ggml-model-q4_0.bin 3` for q4_1

And that's it, youre done!

No error checking for stuff like that in the script isn't very good but everything is going so fast right now that stuff isn't very polished :smile: 

There is a better, more user-friendly script currently in the pipeline here : https://github.com/ggerganov/llama.cpp/pull/545

But it hasn't been properly tested yet to produce correct output. So probably for the best to use the old script until then. But testing the new script is already an option if you cannot make the old one work.

**Reply by MarcGach** [reply]
Thx for your reply
Yes, I have the correct path for three files.
I still get the same error that cannot open the file:

new command : python3 convert-pth-to-ggml.py models/7B 1

PermissionError: [Errno 13] Permission denied: 'models/7B/ggml-model-f16.bin'

I don't quite understand the error.

**Reply by prusnak** [reply]
what does `ls -la models/7B/` say?

**Reply by MarcGach** [reply]
Thx for your help
ls -la models/7B/
total 13161096
drwxr-xr-x 2 root root        4096 mars  27 15:31 .
drwxr-xr-x 3 root root        4096 mars  27 17:12 ..
-rwxr-xr-x 1 root root         100 mars  24 15:45 checklist.chk
-rwxr-xr-x 1 root root 13476939516 mars  24 15:47 consolidated.00.pth
-rwxr-xr-x 1 root root         101 mars  24 15:47 params.json

**Reply by prusnak** [reply]
Are you running the script as root? because the models/7B folder is only writable by root in your setup.

**Reply by MarcGach** [reply]
I'm not running in root cause with sudo, it doesn't work.
I opened the permissions of the file with chmod 777

**Reply by MarcGach** [reply]
So finally how can I get this file "ggml-model-f16.bin" ?

**Reply by miningnome** [reply]
using `python3 convert-pth-to-ggml.py models/the-model 1`
